main.go

```diff
@@ -69,26 +77,62 @@ func run(cmd *exec.Cmd) error {
 	return nil
 }
 
-func kubectl(m interface{}, args ...string) *exec.Cmd {
+func kubectl(m interface{}, args ...string) (cmd *exec.Cmd, cleanupFunc func(), err error) {
```

Comment:
How about this. Just pass the kubectl function a kubeconfig path?

```go
func kubectl(m interface{}, kubeconfig string, args ...string) (*exec.Cmd, error) 
```

Then create a helper

```go
func kubeconfigPath(m interface{}) (string, func(), error)
```

Then blocks like this

```go
	cmd := kubectl(m, "apply", "-f", "-")
	cmd.Stdin = strings.NewReader(d.Get("content").(string))
	if err := run(cmd); err != nil {
		return err
	}

	stdout := &bytes.Buffer{}
	cmd = kubectl(m, "get", "-f", "-", "-o", "json")
	cmd.Stdin = strings.NewReader(d.Get("content").(string))
	cmd.Stdout = stdout
	if err := run(cmd); err != nil {
		return err
	}
```

Become

```go
	kubeconfig, cleanup, err := kubeconfigPath(m)
	if err != nil {
		return fmt.Errorf("determining kubeconfig: %v", err)
	}
	defer cleanup()

	cmd := kubectl(m, kubeconfig, "apply", "-f", "-")
	cmd.Stdin = strings.NewReader(d.Get("content").(string))
	if err := run(cmd); err != nil {
		return err
	}

	stdout := &bytes.Buffer{}
	cmd = kubectl(m, kubeconfig, "get", "-f", "-", "-o", "json")
	cmd.Stdin = strings.NewReader(d.Get("content").(string))
	cmd.Stdout = stdout
	if err := run(cmd); err != nil {
		return err
	}
```